sklearn/semi_supervised/_self_training.py

```diff
@@ -270,6 +276,17 @@ def fit(self, X, y):
         self.classes_ = self.base_estimator_.classes_
         return self
 
+    def _base_estimator_call(self, function_name, *args):
+        check_is_fitted(self)
+        valid_args = self._validate_data(
+            *args, accept_sparse=["csr", "csc", "lil", "dok"], reset=False
+        )
```

Comment:
For now, I think we can go with this since `fit` also uses `force_all_finite=True`.

As a follow up, if `base_estimator_` supports missing values, then the check would be too restrictive. The work around would be to use the `allow_nan` estimator tag on `self.base_estimator_` and configure `_validate_data` accordingly.

Reply:
I agree, it would also make sense to let dataframe pass through for base estimators who can use the features names internally for instance.

But let's do that in a later PR to keep this one focused on the original matter.

Reply:
In retrospect, I find it more intuitive and more consistent with your own PRs to just call `self._check_feature_names(X, reset=False)` instead of nesting calls to `_validate_data`.